### PR TITLE
Issue #12501: Create CI task to lint github actions

### LINKED ISSUE #12501: Create CI task to lint github actions
Noticed during work on #12435, github actions/workflows have a specific format and syntax that can be validated by [actionlint](https://github.com/rhysd/actionlint). While we will have yamllint in place after #12435, actionlint goes a few steps further for github action/workflows (only), to check things like:
 - deprecated commands
 - permissions
 - workflow names
 - job dependencies
 - contextual typing
 - shellcheck integration for scripts in `run`

Checking deprecated commands saves a lot of time from digging through workflow logs to find them. See a list of all checks available at https://github.com/rhysd/actionlint/blob/main/docs/checks.md.

Since we are increasingly favoring github actions and workflows for CI, extra validation in place will help to ease maintainence in the long term.

### PR DESCRIPTION
**Pull request by nrmancuso**
closes #12501

Diff Regression config: https://gist.githubusercontent.com/nrmancuso/858f3ef5e63e12540b71a3e514acdf52/raw/b9e3d229bb0e226cfb11b98e5e3c26895c26faa7/config.xml

### CODE REVIEW

**Review by romani — APPROVED** [review]
Ok

**Comment by romani** [comment]
Can we use https://github.com/rhysd/actionlint/blob/main/docs/usage.md#ignore-some-errors to start merging changes and do improvement step by step

**Comment by nrmancuso** [comment]
@romani all is fixed, I just need to push changes for yamllint errors

**Comment by nrmancuso** [comment]
Github, generate site

**Comment by nrmancuso** [comment]
Github, generate report

File: .github/workflows/no_old_refs.yml [hunk 1]
```diff
@@ -40,15 +40,16 @@ jobs:
           LINKED_ISSUES_FORMATTED=/tmp/linked_issues
           CHECKSTYLE_ISSUE_URL="https:\/\/github.com\/checkstyle\/checkstyle\/issues\/"
           if [ -n "$LINKED_ISSUES" ]; then
+            # shellcheck disable=SC2001
             echo $LINKED_ISSUES | sed -e 's/,/\n/g' >> $LINKED_ISSUES_FORMATTED
```
Comment (nrmancuso) [inline .github/workflows/no_old_refs.yml:44]: >`SC2001: See if you can use ${variable//search/replace} instead.`

Bash expansion/ replacement here made script harder to read, IMO

**Comment by nrmancuso** [comment]
Tested diff report and site generation, see above.

**Review by romani — APPROVED** [review]

**Comment by romani** [comment]
travis and circleci are restarted.

**Review by rnveach — APPROVED** [review]

**Pull request merged**